append_lifecycle.go

```diff
@@ -524,7 +525,10 @@ func (o AppendOptions) CheckpointPublisher(lr LogReader, httpClient *http.Client
 
 		cp, err = wg.Witness(ctx, cp)
 		if err != nil {
-			return nil, err
+			if !o.witnessOpts.FailOpen {
```

Comment:
This should have metrics so that operators can safely determine when they are able to switch modes.

Reply:
Good idea, done